client/src/panels/panel_declarations/services/provided_services_list.js

```diff
@@ -6,25 +6,80 @@ import {
   InfographicPanel,
   declare_panel,
 } from "../shared.js";
+import { FancyUL } from "../../../components";
 
 const { text_maker, TM } = create_text_maker_component(text);
 
-const ProvidedServicesListPanel = ({ panel_args }) => (
-  <div>
-    <TM
-      k="list_of_provided_services_desc"
-      args={{
-        subject_name: panel_args.subject.name,
-        num_of_services: panel_args.services.length,
-      }}
-    />
-    {_.map(panel_args.services, ({ name, id, org_id }) => (
-      <a key={id} href={`#dept/${org_id}/service-panels/${id}`}>
-        {name}
-      </a>
-    ))}
-  </div>
-);
+class ProvidedServicesListPanel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { query: "" };
+  }
+
+  render() {
+    const { panel_args } = this.props;
+
+    const filtered_services = _.filter(panel_args.services, (service) => {
+      if (
+        _.includes(service.name.toLowerCase(), this.state.query.toLowerCase())
```

Comment:
I personally prefer to declare a variable if object needs to be referenced multiple times like
`const { query } = this.state` or `const { services, subject } = panel_args`. This is optional personal preference though

I think variable name `query` can be a bit more specific also, like `service_query`?

Reply:
Also it'd be a bit cleaner to my taste without any if statements
`_.filter(services, service => _.includes(name) || _.includes(type)`